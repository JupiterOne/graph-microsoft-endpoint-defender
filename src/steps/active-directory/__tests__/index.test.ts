// import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';
// import {
//   entities,
//   TargetEntities,
//   fetchAccount,
//   fetchMachines,
//   buildMachineManagesDevicesRelationships,
//   fetchLogonUsers,
//   fetchFindings,
//   buildFindingIsCveRelationships,
// } from '..';
// import { IntegrationConfig } from '../../../../src/config';
// import { Recording, setupProjectRecording } from '../../../../test/recording';

// const config: IntegrationConfig = {
//   clientId: 'a8626c1e-191d-4e8f-9cbc-a4a6e85104ac',
//   clientSecret: 'L3X8Q~BO57QkoAsXMSkevrmVR2qiNh.qKEuzucAt',
//   tenant: '5a721b05-53ed-4ed9-be02-aed28f11edbd',
//   isDefenderApi: false,
// };

// const insufficientPermissionsDirectoryConfig: IntegrationConfig = {
//   clientId: 'clientId',
//   clientSecret: 'clientSecret',
//   tenant: '5a721b05-53ed-4ed9-be02-aed28f11edbd',
//   isDefenderApi: false,
// };
// // See test/README.md for details
// let recording: Recording;
// afterEach(async () => {
//   recording ? await recording.stop() : null;
// });

// describe('fetchAccount', () => {
//   it('Should create an account entity correctly when the account has the correct permissions', async () => {
//     const context = createMockStepExecutionContext({ instanceConfig: config });
//     await fetchAccount(context);

//     const accountEntities = context.jobState.collectedEntities;

//     expect(accountEntities.length).toBe(1);
//     expect(accountEntities).toMatchGraphObjectSchema({
//       _class: entities.ACCOUNT._class,
//     });
//     expect(accountEntities).toMatchSnapshot('accountEntitiesSuccessful');
//   });

//   it('Should error when there are errors attempting to get organization data', () => {
//     const context = createMockStepExecutionContext({
//       instanceConfig: insufficientPermissionsDirectoryConfig,
//     });
//     expect(fetchAccount(context)).rejects;
//   });
// });

// describe('fetchMachines', () => {
//   it('Should create an account entity correctly when the machine has the correct permissions', async () => {
//     recording = setupProjectRecording({
//       directory: __dirname,
//       name: 'fetchMachines',
//     });

//     const context = createMockStepExecutionContext({ instanceConfig: config });
//     await fetchMachines(context);

//     const machineEntities = context.jobState.collectedEntities;

//     expect(machineEntities.length).toBe(0);
//     expect(machineEntities).toMatchGraphObjectSchema({
//       _class: entities.MACHINE._class,
//     });
//     expect(machineEntities).toMatchSnapshot('machineEntitiesSuccessful');
//   });
// });

// describe('machineManagesDevicesRelationships', () => {
//   it('Should create an machine entity correctly when  machine manages devices relationships has the correct permissions', async () => {
//     const context = createMockStepExecutionContext({ instanceConfig: config });
//     await buildMachineManagesDevicesRelationships(context);

//     const machineDeviceEntities = context.jobState.collectedEntities;

//     expect(machineDeviceEntities.length).toBe(0);
//     expect(machineDeviceEntities).toMatchGraphObjectSchema({
//       _class: entities.MACHINE._class,
//     });
//     expect(machineDeviceEntities).toMatchSnapshot(
//       'machineDeviceEntitiesSuccessful',
//     );
//   });
// });

// describe('fetchLogonUsers', () => {
//   it('Should create an User entity correctly when log on User has the correct permissions', async () => {
//     recording = setupProjectRecording({
//       directory: __dirname,
//       name: 'fetchLogonUsers',
//     });

//     const context = createMockStepExecutionContext({ instanceConfig: config });
//     await fetchLogonUsers(context);
//     const logOnUserEntities = context.jobState.collectedEntities;
//     expect(logOnUserEntities.length).toBe(0);
//     expect(logOnUserEntities).toMatchGraphObjectSchema({
//       _class: entities.USER._class,
//     });
//     expect(logOnUserEntities).toMatchSnapshot('logOnUserEntitiesSuccessful');
//   });
// });

// describe('fetchFindings', () => {
//   it('Should create a Vulnerability finding entity correctly when user has the correct permissions', async () => {
//     recording = setupProjectRecording({
//       directory: __dirname,
//       name: 'fetchFindings',
//     });

//     const context = createMockStepExecutionContext({ instanceConfig: config });
//     await fetchFindings(context);

//     const vulnerabilityFindingEntities = context.jobState.collectedEntities;

//     expect(vulnerabilityFindingEntities.length).toBe(0);
//     expect(vulnerabilityFindingEntities).toMatchGraphObjectSchema({
//       _class: entities.FINDING._class,
//     });
//     expect(vulnerabilityFindingEntities).toMatchSnapshot(
//       'vulnerabilityFindingEntitiesSuccessful',
//     );
//   });
// });

// describe('CveRelationshipsFindings', () => {
//   it('Should create a Vulnerability finding entity correctly when there is a CVE relationship', async () => {
//     const context = createMockStepExecutionContext({ instanceConfig: config });
//     await buildFindingIsCveRelationships(context);

//     const cveFindingEntities = context.jobState.collectedEntities;

//     expect(cveFindingEntities.length).toBe(0);
//     expect(cveFindingEntities).toMatchGraphObjectSchema({
//       _class: TargetEntities.CVE._class,
//     });
//     expect(cveFindingEntities).toMatchSnapshot('cveFindingEntitiesSuccessful');
//   });
// });
