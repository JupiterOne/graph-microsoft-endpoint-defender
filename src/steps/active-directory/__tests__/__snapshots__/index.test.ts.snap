// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchAccount Should create an account entity correctly when the account has the correct permissions: accountEntitiesSuccessful 1`] = `
Array [
  Object {
    "_class": Array [
      "Account",
    ],
    "_key": "microsoft_defender_account-local-integration-instance",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "organization": Object {
            "assignedPlans": Array [
              Object {
                "assignedDateTime": "2022-05-25T15:17:24Z",
                "capabilityStatus": "Enabled",
                "service": "Adallom",
                "servicePlanId": "932ad362-64a8-4783-9106-97849a1a30b9",
              },
              Object {
                "assignedDateTime": "2022-05-25T15:17:25Z",
                "capabilityStatus": "Enabled",
                "service": "AADPremiumService",
                "servicePlanId": "41781fb2-bc02-4b7c-bd55-b576c07bb09d",
              },
              Object {
                "assignedDateTime": "2022-05-25T15:17:25Z",
                "capabilityStatus": "Enabled",
                "service": "AADPremiumService",
                "servicePlanId": "eec0eb4f-6444-4f95-aba0-50c24d67f998",
              },
              Object {
                "assignedDateTime": "2022-05-25T15:17:24Z",
                "capabilityStatus": "Enabled",
                "service": "MultiFactorService",
                "servicePlanId": "8a256a2b-b617-496d-b51b-e76466e88db0",
              },
              Object {
                "assignedDateTime": "2022-05-17T07:50:29Z",
                "capabilityStatus": "Enabled",
                "service": "SCO",
                "servicePlanId": "1689aade-3d6a-4bfc-b017-46d2672df5ad",
              },
              Object {
                "assignedDateTime": "2022-05-17T07:50:29Z",
                "capabilityStatus": "Enabled",
                "service": "exchange",
                "servicePlanId": "113feb6c-3fe4-4440-bddc-54d774bf0318",
              },
              Object {
                "assignedDateTime": "2022-05-17T07:50:29Z",
                "capabilityStatus": "Enabled",
                "service": "WindowsDefenderATP",
                "servicePlanId": "871d91ec-ec1a-452b-a83f-bd76c7d770ef",
              },
            ],
            "businessPhones": Array [
              "9493457906",
            ],
            "city": "hyderabad",
            "country": null,
            "countryLetterCode": "IN",
            "createdDateTime": "2022-05-17T07:48:01Z",
            "deletedDateTime": null,
            "directorySizeQuota": Object {
              "total": 50000,
              "used": 188,
            },
            "displayName": "learning",
            "id": "5a721b05-53ed-4ed9-be02-aed28f11edbd",
            "isMultipleDataLocationsForServicesEnabled": null,
            "marketingNotificationEmails": Array [],
            "onPremisesLastSyncDateTime": null,
            "onPremisesSyncEnabled": null,
            "postalCode": "502032",
            "preferredLanguage": "en",
            "privacyProfile": null,
            "provisionedPlans": Array [
              Object {
                "capabilityStatus": "Enabled",
                "provisioningStatus": "Success",
                "service": "Adallom",
              },
              Object {
                "capabilityStatus": "Enabled",
                "provisioningStatus": "Success",
                "service": "SCO",
              },
              Object {
                "capabilityStatus": "Enabled",
                "provisioningStatus": "Success",
                "service": "exchange",
              },
              Object {
                "capabilityStatus": "Enabled",
                "provisioningStatus": "Success",
                "service": "exchange",
              },
            ],
            "securityComplianceNotificationMails": Array [],
            "securityComplianceNotificationPhones": Array [],
            "state": "telangana",
            "street": "test",
            "technicalNotificationMails": Array [
              "sacumentest@Learning232.onmicrosoft.com",
            ],
            "tenantType": "AAD",
            "verifiedDomains": Array [
              Object {
                "capabilities": "Email, OfficeCommunicationsOnline",
                "isDefault": true,
                "isInitial": true,
                "name": "defendertest1.onmicrosoft.com",
                "type": "Managed",
              },
            ],
          },
        },
      },
    ],
    "_type": "microsoft_defender_account",
    "createdOn": undefined,
    "defaultDomain": "defendertest1.onmicrosoft.com",
    "displayName": "Local Integration",
    "id": "5a721b05-53ed-4ed9-be02-aed28f11edbd",
    "name": "learning",
    "organizationName": "learning",
    "verifiedDomains": Array [
      "defendertest1.onmicrosoft.com",
    ],
  },
]
`;
